src/main/java/com/thealgorithms/datastructures/heaps/LeonardoHeap.java

```diff
@@ -0,0 +1,225 @@
+                if (currentLeonardoLevel > 1) {
+                    // compare child and swap
+
+                    int indexOfRightChild = rootNodeIndices.get(j) - 1; // right child is of level n-2
+                    int indexOfLeftChild = rootNodeIndices.get(j) - 1 - LeonardoNumber.leonardoNumber(currentLeonardoLevel - 2);
+                    if (leonardoHeap.get(prevRootNodeIndex).compareTo(leonardoHeap.get(indexOfRightChild)) > 0 && leonardoHeap.get(prevRootNodeIndex).compareTo(leonardoHeap.get(indexOfLeftChild)) > 0) {
+                        swap(prevRootNodeIndex, currentRootNodeIndex);
+                        rootNodeIndexForHeapify = prevRootNodeIndex;
+                        leonardoTreeLevelforHeapify = currentLeonardoTreeLevels[j - 1];
+                        swaped = true;
+                    } else {
+                        maxHeapifyLeonardoTree(currentRootNodeIndex, currentLeonardoLevel);
+                        swaped = false;
+                    }
+                } else {
+                    swap(prevRootNodeIndex, currentRootNodeIndex);
+                    rootNodeIndexForHeapify = prevRootNodeIndex;
+                    leonardoTreeLevelforHeapify = currentLeonardoTreeLevels[j - 1];
+                    swaped = true;
+                }
+                j = j - 1;
+                if (j > 0) {
+                    currentRootNodeIndex = rootNodeIndices.get(j);
+                    prevRootNodeIndex = rootNodeIndices.get(j - 1);
+                } else {
+                    // j = 0 reached the left most tree
+                    break;
+                }
+            }
+
+            if (swaped) {
+                maxHeapifyLeonardoTree(rootNodeIndexForHeapify, leonardoTreeLevelforHeapify);
+                swaped = false;
+            }
+        }
+
+        maxHeapifyLeonardoTree(rootNodeIndexForHeapify, leonardoTreeLevelforHeapify); // In case of insert and no swap.
+    }
+
+    private int getRightMostTree() {
+        // Isolate the rightmost set bit
+        int isolatedBit = leonardoLevelTracker & -leonardoLevelTracker;
+        int position = 0;
+
+        while (isolatedBit > 1) {
+            isolatedBit >>= 1;
+            position++;
+        }
+
+        return position;
+    }
+
+    private static ArrayList<Integer> findConsecutiveLeonardoTreeIndices(int num) {
+        int prevOneIndex = -1;
+        int currentLevel;
+
+        ArrayList<Integer> answer = new ArrayList<Integer>();
+        answer.add(-1);
+        answer.add(-1);
+
+        for (int i = 0; num > 0; i++) {
+            currentLevel = num & 1;
+            if (currentLevel == 1) {
+                if (prevOneIndex != -1) {
+                    answer.set(0, prevOneIndex);
+                    answer.set(1, i);
+                }
+                prevOneIndex = i;
+            } else {
+                prevOneIndex = -1;
+            }
+            num >>>= 1;
+        }
+        return answer;
+    }
```

Comment:
This should be move to a separate class.

Reply:
I can create a new class for this, probably with the name of `LeonardoTreeHelper.java` and add it's respective test file. 

The other thing that can be done is that we can move this function to one of the bit manipulation class? Since all this function is doing is returning first pair of indixes of consecutive 1 bits in an integer. 

Which among these seems the right option @vil02?